builtin/providers/aws/resource_aws_codecommit_trigger.go

```diff
@@ -0,0 +1,170 @@
+							ForceNew: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+
+						"events": &schema.Schema{
+							Type:     schema.TypeList,
+							Required: true,
+							ForceNew: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceAwsCodeCommitTriggerCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).codecommitconn
+	region := meta.(*AWSClient).region
+
+	//	This is a temporary thing - we need to ensure that CodeCommit is only being run against us-east-1
+	//	As this is the only place that AWS currently supports it
+	if region != "us-east-1" {
+		return fmt.Errorf("CodeCommit can only be used with us-east-1. You are trying to use it on %s", region)
+	}
+
+	// Expand the "trigger" set to aws-sdk-go compat []*codecommit.RepositoryTrigger
+	rawTriggers := d.Get("trigger").(*schema.Set).List()
+	triggers := make([]*codecommit.RepositoryTrigger, 0, len(rawTriggers))
+	for _, cors := range rawTriggers {
+		corsMap := cors.(map[string]interface{})
+		r := &codecommit.RepositoryTrigger{}
+		for k, v := range corsMap {
+			if k == "custom_data" {
+				r.CustomData = aws.String(v.(string))
+			} else if k == "destination_arn" {
+				r.DestinationArn = aws.String(v.(string))
+			} else if k == "name" {
+				r.Name = aws.String(v.(string))
+			} else {
+				vMap := make([]*string, len(v.([]interface{})))
+				for i, vv := range v.([]interface{}) {
+					str := vv.(string)
+					vMap[i] = aws.String(str)
+				}
+				switch k {
+				case "branches":
+					r.Branches = vMap
+				case "events":
+					r.Events = vMap
+				}
+			}
+		}
+		triggers = append(triggers, r)
+	}
+
+	input := &codecommit.PutRepositoryTriggersInput{
+		RepositoryName: aws.String(d.Get("repository_name").(string)),
+		Triggers:       triggers,
+	}
+
+	resp, err := conn.PutRepositoryTriggers(input)
+	if err != nil {
+		return fmt.Errorf("Error creating CodeCommit Trigger: %s", err)
+	}
+
+	log.Printf("[INFO] Code Commit Trigger Created %s", resp)
+
+	d.SetId(d.Get("repository_name").(string))
+
+	return resourceAwsCodeCommitTriggerRead(d, meta)
+}
+
+func resourceAwsCodeCommitTriggerRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).codecommitconn
+
+	input := &codecommit.GetRepositoryTriggersInput{
+		RepositoryName: aws.String(d.Id()),
+	}
+
+	resp, err := conn.GetRepositoryTriggers(input)
+	if err != nil {
+		return fmt.Errorf("Error reading CodeCommit Trigger: %s", err.Error())
+	}
+
+	log.Printf("[INFO] CodeCommit Trigger READ %s", resp)
+
+	return nil
+}
+
+func resourceAwsCodeCommitTriggerDelete(d *schema.ResourceData, meta interface{}) error {
+
+	conn := meta.(*AWSClient).codecommitconn
+
+	log.Printf("[DEBUG] Deleting Trigger: %q", d.Id())
+
+	input := &codecommit.PutRepositoryTriggersInput{
+		RepositoryName: aws.String(d.Get("repository_name").(string)),
+		Triggers:       []*codecommit.RepositoryTrigger{},
```

Comment:
Oh i see what you are doing - to remove the triggers, you need to Put an empty list